### ISSUE TITLE
Copy on select - Cannot disable : Preference is ignored

### ISSUE THREAD

**Issue by AtomicRobotMan0101** [hluk/CopyQ#2455]
Before creating a new issue, see list of [known
issues](https://copyq.readthedocs.io/en/latest/known-issues.html).

**Describe the bug**
The "Copy on Select" option is being ignored.  One cannot turn it off.

When CopyQ is closed, this behaviour does not occur.

**To Reproduce**

File --> Preferences --> General --> Clipboard Manipulation --> (2) Store text selected using mouse
This option is being ignored.
Whether on, or off, the content of any mouse highlight is added to the clipboard.

**Expected behavior**

**Screenshots**

![copyq option ignored](https://github.com/hluk/CopyQ/assets/64177332/cb6c8c79-de6e-42dd-b1d7-7b41d3670701)

**Version, OS and Environment**
Linux, Mint 21.2 Cinnamon.  Here is my `inxi -Fzxxx` --> https://termbin.com/39swe

```
$ copyq --version
CopyQ Clipboard Manager 7.1.0~jammy
Qt: 5.15.3
KNotifications: 5.92.0
Compiler: GCC
Arch: x86_64-little_endian-lp64
OS: Linux Mint 21.2
```

**Additional context**
Nothing I can think of. Problem is quite reproducible.

**Comment by hluk** [comment]
> File --> Preferences --> General --> Clipboard Manipulation --> (2) Store text selected using mouse
> This option is being ignored.
> Whether on, or off, the content of any mouse highlight is added to the clipboard.

The option is for storing the text selected by mouse in other apps in CopyQ - i.e. storing X11 "primary" selection (in addition to "clipboard"). It has no relation to selecting items in CopyQ.

**Comment by AtomicRobotMan0101** [comment]
I thought this too, but, when copyq is closed, the behaviour stops.

When copyq is running, any highlighted text is copied, regardless of the setting.

It's most curious.

**Comment by hluk** [comment]
Ah, sorry I misunderstood the issue (for some reason I thought it is related to selecting items in CopyQ history).

If I understand correctly, even if the "(2) Store text selected using mouse" is disabled, CopyQ still stores texts selected by mouse in its history. Is that correct?

Do you use an "Automatic" command that would affect the behavior? If you open the Command dialog (F6 shortcut), the "Automatic" commands are listed with a small "clipboard" icon.

Also, some apps (especially terminal emulators) can store mouse selected text also to clipboard. Can you check if that is the case?

**Comment by AtomicRobotMan0101** [comment]
I must most humbly apologise.

It was my settings in Gnome Terminal, Firefox (the Swift Selection Search add-on), Thunderbird (same style plugin) and my Geany and Xed text editors.

CopyQ is working perfectly, as it always has been. The error is all mine.

BTW, thank you for such a fantastic tool. I lodged the "bug" as I find it so terribly useful !! 😺

**Comment by hluk** [comment]
No need to apologize. I think the problem is actually that some of the in-app texts are a bit unclear. Let me know if you have any ideas to improve them. Thanks for the report.

**Comment by jashment** [comment]
@hluk I hate to comment when this has already been closed but this is the exact same issue that I am having. As far as I am able to tell, the highlighted text gets copied in any app that I am using. As far as I am able to tell, I don't have any global settings that would be causing this or any automatic commands either. I am using Ubuntu Budgie:

```
Budgie Welcome: 0.21.4
Budgie Desktop 10.6.1-1ubuntu3
Ubuntu Budgie 22.04 jammy
```

I don't have Store text selected using mouse or run automatic commands on selection checked either.

Any suggestions?

**Comment by AtomicRobotMan0101** [comment]
When I made the initial assumption it was CopyQ, I simply believed the checkbox (image below, selection 2) was not operating properly (i.e. the Qt element was there, but no action(?) attached to it).

Normally, it would not bother me, but I'm of the habit of highlighting lines and chunks of text/code before deleting it. This caused the highlighted code to be copied into the scratchpad (?name?)

I was wrong.

_It turned out to be the various plugins I run all over the place._

**Gnome Terminal** has copy on highlight by default. In **Firefox** I use **Swift Selection Search** (https://github.com/CanisLupus/swift-selection-search) (HIGHLY recommended!) and in **other programs similar tools**.

I simply forgot (or didn't think) to check all these extra places. I TRY to keep my play/work/dev space clean, but tools are often just rammed into place and forgotten about.

In my searching to "solve" this problem, I did not find a place or option in Debian/Ubuntu/Mint where the copy-on-select was a universal OS-like option.

Soz I can't offer further wisdom....


![copyq](https://github.com/hluk/CopyQ/assets/64177332/a27d89d1-46c5-427b-815d-f8a86963fc4e)
